Flickr/Sources/FlickrAppVCs/PhotoViewerVC.swift

```diff
@@ -0,0 +1,161 @@
+        let stackViewConstraints = setupStackView()
+        NSLayoutConstraint.activate(stackViewConstraints)
+    }
+
+    private func setupStackView() -> [NSLayoutConstraint] {
+        let stackView = UIStackView()
+        stackView.configureView { stackView in
+            stackView.axis = .horizontal
+            stackView.alignment = .fill
+            stackView.distribution = .fill
+            stackView.spacing = 20
+        }
+        self.stackView = stackView
+        let labelConstraints = returnLabel(stackView: stackView)
+        NSLayoutConstraint.activate(labelConstraints)
+        let iconConstraints = returnIcon(stackView: stackView)
+        NSLayoutConstraint.activate(iconConstraints)
+        view.addSubview(stackView)
+
+        guard let imageView = imageView else {
+            return []
+        }
+        return [
+            stackView.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 15.0),
+            stackView.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -15.0),
+            stackView.topAnchor.constraint(equalTo: imageView.bottomAnchor, constant: 20.0)
+        ]
+    }
+
+    private func returnLabel(stackView: UIStackView) -> [NSLayoutConstraint] {
+        let label = UILabel()
+        label.configureView { label in
+            label.text = imageTitle
+            label.font = label.font.withSize(20)
+            label.clipsToBounds = false
+        }
+        stackView.addArrangedSubview(label)
+
+        return [
+            label.topAnchor.constraint(equalTo: stackView.topAnchor),
+            label.trailingAnchor.constraint(equalTo: stackView.trailingAnchor, constant: -60)
+        ]
+    }
+
+    private func returnIcon(stackView: UIStackView) -> [NSLayoutConstraint] {
+        let enabledButton = UIButton()
+        enabledButton.configureView { button in
+            button.imageView?.contentMode = .scaleAspectFit
+        }
+        let filledIcon = filledHeartIcon
+        enabledButton.isHidden = true
+        enabledButton.setImage(filledIcon, for: .normal)
+        enabledButton.addTarget(self, action: #selector(clickedFavorite), for: .touchUpInside)
+        favoriteEnabledButton = enabledButton
+
+        let disabledButton = UIButton()
+        disabledButton.configureView { button in
+            button.imageView?.contentMode = .scaleAspectFit
+        }
+        let outlinedIcon = outlinedHeartIcon
+        disabledButton.setImage(outlinedIcon, for: .normal)
+        disabledButton.isHidden = false
+        disabledButton.addTarget(self, action: #selector(clickedFavorite), for: .touchUpInside)
+        favoriteDisabledButton = disabledButton
+
+        stackView.addArrangedSubview(enabledButton)
+        stackView.addArrangedSubview(disabledButton)
+
+        return [
+            enabledButton.topAnchor.constraint(equalTo: stackView.topAnchor),
+            disabledButton.topAnchor.constraint(equalTo: stackView.topAnchor)
+        ]
```

Comment:
Why do you need two buttons for the favorites button? can you not achieve this with just one button?
Also ensure that each function is responsible for setting up only 1 component of the view.